### ISSUE TITLE
Beyond-HD returning extra results

### ISSUE THREAD

**Issue by newadventure079** [Jackett/Jackett#12631]
### Have you checked our Troubleshooting page for your issue?

- [X] I have checked the Troubleshooting page

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Environment

```
2021-12-02 19:53:03	Info	Using HTTP Client: HttpWebClient2
2021-12-02 19:53:03	Info	Using FlareSolverr: No
2021-12-02 19:53:03	Info	Using proxy: Disabled
2021-12-02 19:53:03	Info	App config/log directory: /config/Jackett
2021-12-02 19:53:03	Info	ThreadPool MaxThreads: 32767 workerThreads, 1000 completionPortThreads
2021-12-02 19:53:03	Info	Running in Docker: Yes (image build: v0.20.83-ls51)
2021-12-02 19:53:03	Info	File /etc/issue: Ubuntu 18.04.6 LTS \n \l
2021-12-02 19:53:03	Info	Jackett variant: CoreLinuxAmdx64
2021-12-02 19:53:03	Info	OS version: Unix 4.4.180.0 (64bit OS) (64bit process)
2021-12-02 19:53:03	Info	Environment version: 6.0.0 (/app/Jackett/)
2021-12-02 19:53:02	Info	Starting Jackett v0.20.83
```
### Description

Searching for `Orange Is the New Black S06` using BHD returns results for additional seasons

The same search when using the BHD website works fine `https://beyond-hd.me/torrents?search=Orange+Is+the+New+Black+S06&doSearch=Search`

There's only 5 results returned when you view the search on the BHD website but 28 results are shown in Jackett, all extraneous results like S01, S02, etc

### Logged Error Messages

`Manual search in beyond-hd-api for Orange Is the New Black S06 => Found 28 releases`

### Screenshots

![Screen Shot 2021-12-02 at 7 47 50 PM](https://user-images.githubusercontent.com/84156407/144536705-7cbf610e-3285-4af4-99c7-b016346eb321.jpg)

**Comment by ilike2burnthing** [comment]
Please post an enhanced log - https://github.com/Jackett/Jackett/wiki/Troubleshooting#how-to-provide-a-jackett-enhanced-log-for-a-ticket

I think the season string is being removed.

**Comment by newadventure079** [comment]
Logs are attached

[log.txt](https://github.com/Jackett/Jackett/files/7651518/log.txt)

**Comment by ilike2burnthing** [comment]
@PNWebster is the season being stripped out by:
https://github.com/Jackett/Jackett/blob/aecc9f75d5aa21654bd97d5b687d1e33fe7a947d/src/Jackett.Common/Indexers/BeyondHDAPI.cs#L100-L101

Does it need to be?

**Comment by PNWebster** [comment]
@ilike2burnthing that line is not the issue.

What is happening is that the ApiSearch method ToTorznabQuery is applying regex to find a season match. The "S06" value is matched, and the torznab query is modified. 
The search string is set to "Orange Is The New Black", and the Season is set to "6".

But the queryType is not set to "tvSearch". So it does not enter the TV search logic I created for the BHD API, and instead just falls back to a generic search based on only the search string, which has had the season information removed. So it finds all torrents for "Orange Is The New Black".

Sidenote: I was first unable to reproduce because I tested with "orange is the new black s06". But the lowercase s in "s06" did not cause the ApiSearch regex to match season.

So, I'm not sure what the correct behaviour here should be. Few options:
- Should ApiSearch set the queryType to "tvsearch" when it finds a season?
- Should BHD API indexer assume a TV search if season has a value?
- Should BHD API indexer re-append the season if it has a value but the searchtype was not TV?
